.github/workflows/pr_closed.yml

```diff
@@ -15,7 +15,19 @@ jobs:
         with:
           ref: gh-pages
 
+      - name: Determine if cleaning is necessary
+        id: should-clean
+        run: |
+          if [ -d "_preview/${{ github.event.pull_request.number }}" ]; then
+            echo "PR preview exists, cleaning"
+            echo "clean=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "PR preview does not exist, skipping"
+            echo "clean=false" >> "$GITHUB_OUTPUT"
+          fi
+
       - name: Remove PR preview
+        if: ${{ steps.should-clean.outputs.clean == 'true' }}
         run: rm -rf _preview/${{ github.event.pull_request.number }}
```

Comment:
This could not have been the cause of the error, because `rm -rf` works even if the given path does not exist.

<img width="282" alt="Screenshot 2023-04-15 at 1 06 11 AM" src="https://user-images.githubusercontent.com/16580576/232154827-1b3c2713-aef2-4149-9bfc-888d1850c67c.png">

Reply:
Yep, indeed 🙂 Do you have a different proposal for addressing this with that in mind?

Reply:
This `if` condition should be on the next step, "Push cleaned branch" because that's the one that fails when there is no change to commit.

Reply:
Got it, my thought was that this delete step wasn't necessary to do if no changes were made (regardless of whether it would succeed or not). Skipping it would also prevent giving the false impression that the removal step was actually removing something too, which could be cause for more confusion.

Reply:
Dhruv and I talked offline about this and it seems I was misunderstanding his point - my assumption was that skipping the delete step would also skip the downstream step (this is how Airflow works). I'm not sure GitHub Actions works this way, so I'll need to double check! If that's the case, then I'll move the conditional to the last step instead, since that _is_ the one we want to make sure is skipped.

Reply:
Yep, I was thinking about this all wrong! I'll need to change the step that gets skipped: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif

Thanks @dhruvkb!